templates/user_profile.html

```diff
@@ -0,0 +1,377 @@
+{% extends "layout.html" %}
+{% block pagetitle %}My profile{% endblock %}
+{% block additionalpageheader %}
+    <style>
+        .move-icon {
+            cursor: move;
+        }
+
+        .delete-icon {
+            cursor: pointer;
+            color: red;
+        }
+    </style>
+    <script>
+        $(document).ready(function () {
+            function toggleSupervisorSelect() {
+                if ($('#is_researcher').is(':checked')) {
+                    $('#supervisor_select').show();
+                    $('#researcher_type_display').show();
+                    $('#max_load_display').show();
+                } else {
+                    $('#supervisor_select').hide();
+                    $('#researcher_type_display').hide();
+                    $('#max_load_display').hide();
+                }
+            }
+
+            // Initial check on page load
+            toggleSupervisorSelect();
+
+            $('#is_researcher').change(function () {
+                toggleSupervisorSelect();
+            });
+        });
+    </script>
+    <script>
+        $(document).ready(function () {
+            $('#addPreferencesButton').click(function () {
+                $('#staticPreferencesTable').hide();
+                $('#editablePreferencesTable').show();
+                $('#addPreferencesButton').hide();
+                $('#savePreferencesButton').show();
+                $('#cancelEditButton').show();
+                $('#addRowButton').show();
+            });
+
+            $('#cancelEditButton').click(function () {
+                $('#editablePreferencesTable').hide();
+                $('#staticPreferencesTable').show();
+                $('#addPreferencesButton').show();
+                $('#savePreferencesButton').hide();
+                $('#cancelEditButton').hide();
+                $('#addRowButton').hide();
+            });
+
+            $('#addRowButton').click(function () {
+                const newRow = `
+                    <tr>
```

Comment:
I strongly recommend you to transform this piece of code as a hidden HTML element and use it as a template. You select the template using JQuery, copy it and set the data before inserting it at the right location.